lib/matplotlib/image.py

```diff
@@ -361,6 +362,12 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                     a_min, a_max = np.int32(0), np.int32(1)
                 if inp_dtype.kind == 'f':
                     scaled_dtype = A.dtype
+                    # Cast to float64
+                    if A.dtype == np.longdouble:
```

Comment:
OK, still not sure what to do about `np.float128` and `np.float96`.  Should these be trapped here as well?

Reply:
Based on the platform version, np.float128 or np.float96 may or may not exist so adding traps for them might not work. Instead, I think casting anything that isn't float32 or float16 to float64 would be a better idea.

Reply:
Really, they aren't even defined?  Yuck....

Reply:
Yes. np.longdouble is platform dependent also. That's why I most likely have to remove the checking for warnings from the test case since the CI runs on Windows which has np.longdouble == 'float64' == 'float'. This causes either excessive warnings which will break all sorts of CI tests (if I choose to show warnings for each cast) or none at all (if I only show warnings on casts from a larger size to smaller).